model-optimizer/mo/utils/ir_reader/restore_graph.py

```diff
@@ -17,7 +20,20 @@
 from mo.utils.ir_engine.ir_engine import IREngine
 from mo.utils.ir_reader.layer_to_class import copy_graph_with_ops, collect_extenders, collect_ops
 from mo.utils.utils import get_mo_root_dir
-from extensions.back.MarkNodesWithShapeValues import MarkNodesWithShapeValues
+
+
+def define_data_type(graph: Graph):
```

Comment:
As discussed please try to do simplified logic: if data_type is not present assume the model is FP32 (which means do not do FP16 conversion)

Reply:
@mvafin, @achetver, can it be a case that someone creates fp16 network using nGraph API so `data_type` is not set?

Reply:
Yes, it can. In this case if user doesn't provide to function `save_restored_graph(graph: Graph, path: str, meta_data, name=None)` data_type using `meta_data` parameter we set data_type to fp32 (this info will not be saved in IR and used only in our internal checks) but save all operations in provided data_type (restored from IR or setted by user)